src/test/java/edu/ksu/canvas/attendance/submitter/CanvasAssignmentAssistantUTest.java

```diff
@@ -0,0 +1,176 @@
+
+        Whitebox.setInternalState(canvasAssignmentAssistant, "assignmentService", assignmentService);
+        Whitebox.setInternalState(canvasAssignmentAssistant, "attendanceSectionService", attendanceSectionService);
+        Whitebox.setInternalState(canvasAssignmentAssistant, "assignmentRepository", assignmentRepository);
+        Whitebox.setInternalState(canvasAssignmentAssistant, "canvasApiWrapperService", canvasApiWrapperService);
+    }
+
+    @Test
+    public void createAssignmentInCanvasHappyPath() throws IOException, AttendanceAssignmentException {
+        when(canvasApiWrapperService.createAssignment(any(), any(), any())).thenReturn(assignmentOptional);
+        when(attendanceSectionService.getSectionsByCourse(COURSE_ID)).thenReturn(sectionList);
+        when(assignmentService.findBySection(any())).thenReturn(attendanceAssignment);
+        when(assignmentRepository.save(attendanceAssignment)).thenReturn(attendanceAssignment);
+
+        canvasAssignmentAssistant.createAssignmentInCanvas(COURSE_ID, attendanceAssignment, oauthToken);
+    }
+
+    @Test
+    public void createAssignmentInCanvasNullReturnError() throws IOException, AttendanceAssignmentException {
+        assignmentOptional = Optional.empty();
+
+        when(canvasApiWrapperService.createAssignment(any(), any(), any())).thenReturn(assignmentOptional);
+
+        try {
+            canvasAssignmentAssistant.createAssignmentInCanvas(COURSE_ID, attendanceAssignment, oauthToken);
+            Assert.fail("Expected AttendanceAssignmentException");
+        } catch (AttendanceAssignmentException e) {
+            Assert.assertEquals(AttendanceAssignmentException.Error.CREATION_ERROR, e.error);
+        }
+    }
+
+    @Test
+    public void editAssignmentInCanvasHappyPath() throws IOException, AttendanceAssignmentException {
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID+"")).thenReturn(assignmentOptional);
+
+        canvasAssignmentAssistant.editAssignmentInCanvas(COURSE_ID, attendanceAssignment, oauthToken);
+        verify(canvasApiWrapperService, times(1)).editAssignment(COURSE_ID.toString(), assignment, oauthToken);
+    }
+
+    @Test
+    public void editAssignmentInCanvasCanvasAssignmentNotFound() throws IOException, AttendanceAssignmentException {
+        assignmentOptional = Optional.empty();
+
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID+"")).thenReturn(assignmentOptional);
+
+        try {
+            canvasAssignmentAssistant.editAssignmentInCanvas(COURSE_ID, attendanceAssignment, oauthToken);
+        } catch(AttendanceAssignmentException exception ) {
+            Assert.assertEquals(AttendanceAssignmentException.Error.NO_ASSIGNMENT_FOUND, exception.error);
+        }
+    }
+
+    @Test
+    public void deleteAssignmentInCanvasHappyPath() throws IOException, AttendanceAssignmentException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.AttendanceAssignmentException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)